lib/mm_code.rb

```diff
@@ -0,0 +1,76 @@
+CONVERSION_CONTAINER = {
+    1 => 'R',
+    2 => 'G',
+    3 => 'Y',
+    4 => 'B',
+    5 => 'M',
+    6 => 'C'
+}
+
+COLOR_CONTAINER = {
+    1 => "\e[91m●\e[0m",
+    2 => "\e[92m●\e[0m",
+    3 => "\e[93m●\e[0m",
+    4 => "\e[94m●\e[0m",
+    5 => "\e[95m●\e[0m",
+    6 => "\e[96m●\e[0m"
+}
+
+class Code
+    attr_reader :value, :matched
+
+    def initialize(new_code)
+        if /[rgybmcRGYBMC]/ === new_code[0]
+            new_code = new_code.join.upcase.split(//)
+            @value = convert(new_code)
+        else
+            @value = new_code
+        end
+    end
+
+    def convert(input = @value)
+        input.map do |v|
+            if v.to_i.positive?
+                CONVERSION_CONTAINER[v.to_i]
+            else
+                CONVERSION_CONTAINER.key(v)
+            end
+        end
+    end
+
+    def convert_to_color(input = @value)
+        input.map do |v|
+            COLOR_CONTAINER[v.to_i]
+        end
+    end
```

Comment:
`convert_to_color` seems to be concerned with coloring (having to do with the UI?). I wouldn't expect a class called `Code` to be concerned with UI. Can we decouple UI from any logic that this class is concerned with?